pkg/crc/cluster/cert-renewal.go

```diff
@@ -2,25 +2,43 @@ package cluster
 
 import (
 	"fmt"
-	"strings"
 	"time"
 
 	crcerrors "github.com/code-ready/crc/pkg/crc/errors"
 	"github.com/code-ready/crc/pkg/crc/logging"
 	"github.com/code-ready/crc/pkg/crc/oc"
 	"github.com/code-ready/crc/pkg/crc/ssh"
+
+	k8scerts "k8s.io/api/certificates/v1beta1"
 )
 
-func waitForPendingCSRs(ocConfig oc.Config, signerName string) error {
+func isPending(csr *k8scerts.CertificateSigningRequest) bool {
+	return len(csr.Status.Conditions) == 0 && len(csr.Status.Certificate) == 0
+}
+
+func approvePendingCSRs(ocConfig oc.Config, expectedSignerName string) error {
 	return crcerrors.RetryAfter(8*time.Minute, func() error {
-		output, _, err := ocConfig.WithFailFast().RunOcCommand("get", "csr")
+		csrs, err := getCSRList(ocConfig, expectedSignerName)
 		if err != nil {
 			return &crcerrors.RetriableError{Err: err}
 		}
-		if strings.Contains(output, "Pending") && strings.Contains(output, signerName) {
-			return nil
+		csrsApproved := false
+		for i := range csrs.Items {
+			csr := &csrs.Items[i]
+			if !isPending(csr) {
+				continue
+			}
+			logging.Debugf("Approving csr %s (signerName: %s)", csr.ObjectMeta.Name, expectedSignerName)
+			_, stderr, err := ocConfig.RunOcCommand("adm", "certificate", "approve", csr.ObjectMeta.Name)
+			if err != nil {
+				return fmt.Errorf("Not able to approve csr (%v : %s)", err, stderr)
+			}
+			csrsApproved = true
+		}
+		if !csrsApproved {
+			return &crcerrors.RetriableError{Err: fmt.Errorf("No Pending CSR with signerName %s", expectedSignerName)}
 		}
-		return &crcerrors.RetriableError{Err: fmt.Errorf("No Pending CSR with signerName %s", signerName)}
+		return nil
 	}, time.Second*5)
```

Comment:
I think this retry option should shift to `getCSRList` function because this function only provide the csr list and to get the list we need to make sure  to wait till `csr` resource is available. With this path now we are retry on the error coming from the `getCSRList` which might be a valid error not because of  of resource not available.

Reply:
The `retry` needs to wait until:
1) a list of CSRs is returned
2) there is at least one pending CSR

If we move the `retry` to `getCSRList`, we can do 1) but not 2). I agree adding a short retry loop in `getCSRList`would probably be good though.

Reply:
@cfergeau Yes I am talking about the 1 where we do get the list of csr. pending logic should  be here because here we are approving the pending csr.

Reply:
I dropped the 'csr' patches from this PR for now so that we can move forward with the other unrelated changes.